tests/integration/integration_test.go

```diff
@@ -50,6 +81,24 @@ func TestEmptyGo(t *testing.T) {
 	})
 }
 
+// Tests emitting many engine events doesn't result in a performance problem.
+func TestEngineEventPerf(t *testing.T) {
```

Comment:
I didn't think of it sooner, but I wonder if there would be advantages to making this a "Benchmark" (in the Go testing package sense), rather than a "Test". I'm guessing we are going to eventually hit flakiness due to machine speed differences, parallel execution, etc, and AFAIK the benchmark stuff was designed to give us knobs to control those kinds of problems. Just worth a look.